### PR TITLE
Bug/1555/wrong display of colored folders in legend

### PR DESCRIPTION
**Pull request by RomanenkoVladimir**
# Wrong display of colored folders in legend

_**Please read the [CONTRIBUTING.md](https://github.com/MaibornWolff/codecharta/blob/main/CONTRIBUTING.md) before opening a PR.**_

closes: #1555

## Description
- The feature was already implemented and tested correctly, however we were tracking elements by $index, which means that angular would not redraw the new information ( we specifically told the DOM that way that this component would not change) -> Removed incorrect tracking
- Fixed incorrect commata placement

### CODE REVIEW

**Comment by RomanenkoVladimir** [comment]
Concerning tests, how would we test for this functionality? 
We already have unit test, however the values were always correct, the issue was manually hindering update propagandation.

File: visualization/app/codeCharta/ui/legendPanel/legendPanel.component.html [hunk 1]
```diff
@@ -25,19 +25,15 @@
...
-		ng-show="$ctrl._viewModel.packageLists"
-		ng-repeat="packageList in $ctrl._viewModel.packageLists track by $index"
-	>
+	<hr ng-if="$ctrl._viewModel.packageLists[0]" />
+	<div class="legend-block" ng-show="$ctrl._viewModel.packageLists" ng-repeat="packageList in $ctrl._viewModel.packageLists">
 		<img ng-src="{{::packageList.colorPixel}}" alt="" />
 		<span
 			class="marked-package {{ $first ? '' : 'set-left-margin' }}"
-			ng-repeat="item in packageList.markedPackages track by $index"
+			ng-repeat="item in packageList.markedPackages"
```
Comment (BridgeAR) [inline visualization/app/codeCharta/ui/legendPanel/legendPanel.component.html:33]:
```suggestion
			ng-repeat="item in packageList.markedPackages track by item.path"
```

**Review by BridgeAR — COMMENTED** [review]
It's a best practice to track by unique identifiers when iterating over objects. We mostly use the full path as identifier and thus, it's good to use this. Otherwise the whole object is stringified and that takes a while.

See https://docs.angularjs.org/api/ng/directive/ngRepeat#tracking-and-duplicates

File: visualization/app/codeCharta/ui/legendPanel/legendPanel.component.html (hunk 1 above)
Comment (RomanenkoVladimir) [reply visualization/app/codeCharta/ui/legendPanel/legendPanel.component.html]: I see, thanks
I changed it to avoid full object stringifying

**Review by BridgeAR — COMMENTED** [review]
LGTM but ideally, we'd add an e2e test to verify the behavior to prevent future regressions :-)

File: visualization/app/codeCharta/ui/legendPanel/legendPanel.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+		legendPanelObject = new LegendPanelObject()
+		fileChooser = new FileChooserPageObject()
+
+		await goto()
+	})
+
+	describe("LegendPanel", () => {
+		it("should highlight a folder and add to legend then react to a changed color", async () => {
+			await fileChooser.openFiles(["./app/codeCharta/ressources/sample1_with_different_edges.cc.json"])
+			await legendPanelObject.open()
```
Comment (BridgeAR) [inline visualization/app/codeCharta/ui/legendPanel/legendPanel.e2e.ts:19]: This does not yet test the output, does it? (Correct display of the file names)

**Review by BridgeAR — APPROVED** [review]
LGTM

**Pull request merged**